projects/sdk/src/lib/silo/Withdraw.ts

```diff
@@ -0,0 +1,128 @@
+import { ContractTransaction } from "ethers";
+import { Token } from "src/classes/Token";
+import { TokenValue } from "src/classes/TokenValue";
+import { BeanstalkSDK } from "../BeanstalkSDK";
+import { DepositCrate, Silo } from "../silo";
+import { sortCratesBySeason } from "../silo.utils";
+
+export class Withdraw {
+  static sdk: BeanstalkSDK;
+
+  constructor(sdk: BeanstalkSDK) {
+    Withdraw.sdk = sdk;
+  }
+
+  /**
+   * Initates a withdraw from the silo. The `token` specified dictates which silo to withdraw
+   * from, and therefore is limited to only whitelisted assets.
+   * Behind the scenes, the `amount` to be withdrawn must be taken from individual
+   * deposits, aka crates. A user's deposits are not summarized into one large bucket, from
+   * which we can withdraw at will. Each deposit is independently tracked, so each withdraw must
+   * calculate how many crates it must span to attain the desired `amount`.
+   * @param token The whitelisted token to withdraw. ex, BEAN vs BEAN_3CRV_LP
+   * @param amount The desired amount to withdraw. Must be 0 < amount <= total deposits for token
+   * @returns Promise of Transaction
+   */
+  async withdraw(token: Token, amount: TokenValue): Promise<ContractTransaction> {
```

Comment:
I think another method name might work better. Right now, you end up calling `sdk.silo.withdraw.withdraw` and it's weird, but other entities build the entity then `.execute()`, e.g. `sdk.silo.buildDeposit` then `deposit.execute()`. Similarly with swaps. Does that pattern not make sense here for some reason?